drivers/counter/counter_dw_timer.c

```diff
@@ -0,0 +1,413 @@
+/*
+ * Copyright (c) 2023 Intel Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT snps_dw_timers
+
+#include <zephyr/device.h>
+#include <zephyr/drivers/counter.h>
+#include <zephyr/spinlock.h>
+#include <zephyr/logging/log.h>
+
+#define COUNTER_DW_RESET_SUPPORT \
+	DT_ANY_INST_HAS_PROP_STATUS_OKAY(resets)
```

Comment:
Hiding parameters in a macro is really bad practice, so either you make `(resets)` explicit or you entirely remove this define.

Reply:
Incorportated the changes, used directly with the property now
#if DT_ANY_INST_HAS_PROP_STATUS_OKAY(resets)
#endif